src/modules/app/app.gateway.ts

```diff
@@ -101,34 +142,61 @@ export class AppGateway implements OnGatewayDisconnect {
         this.server.to(roomId).emit('updateMessage', [newMessage, roomId])
     }
 
-    handleDisconnect(client: Socket) {
-        this.activeUsers.delete(client.id)
-        client.broadcast.emit('activeUsers', [...this.activeUsers.keys()])
+    imageFromBase(image: string) {
+        const [metaData, content] = image.split(',')
+        const buff = Buffer.from(content, 'base64')
+        const fileType = metaData
+            .replace(/.*\//, '')
+            .replace(/;.*/, '')
+
+        return [buff, fileType]
+    }
+
+    saveImage(image: string, clientId: string): string{
+        const [content, fileType] = this.imageFromBase(image)
+        const imagePath = `${ process.env.IMAGES_PATH }\\${ clientId }-${ Math.random()
+            .toString(36)
+            .substr(2, 16) }.${ fileType }`
+            .trim()
+
+        fs.writeFile(`${ imagePath }`, content, () => {})
+
+        return imagePath
+    }
+
+    async handleDisconnect(client: Socket) {
+        await this.appsService.deactivateUser(client.id)
+
+        const activeUsers = await this.appsService.getActiveUsers()
+
+        client.broadcast.emit('activeUsers', activeUsers)
     }
 
-    prepareMessage(messageCall: MessageCall, clientId: string) {
-        const [currentUserName, userAvatar] = this.users.get(clientId)
-        const time = new Date().getTime()
-        const newMessage: Message = {
-            userName: currentUserName,
-            userId: clientId,
-            avatar: userAvatar,
-            message: messageCall.messageContent,
-            image: messageCall.image,
-            date: time
+    async saveUser(socketId: string, avatar: string, userName: string){
+        if(avatar.includes('http://localhost')) {
```

Comment:
in this case You can only have one avatar for every users. Pleas also remove this avatar from git